sympy/solvers/decompogen.py

```diff
@@ -51,6 +52,16 @@ def decompogen(f, symbol):
         result += [f.subs(arg, symbol)] + decompogen(arg, symbol)
         return result
 
+    # ===== Min/Max Functions ===== #
+    if isinstance(f, (Min, Max)):
+        if all(a.has(symbol) for a in f.args):
+            raise TypeError('cannot decompose %s' % f)
+        for i in f.args:
+            if i.has(symbol):
+                arg = i
+        result += [f.subs(arg, symbol)] + decompogen(arg, symbol)
```

Comment:
```suggestion
        args = list(f.args)
        iargs = [i for i, a in enumerate(f.args) if a.has(symbol)]
        if not iargs:
            return result
        dec = []
        for i in iargs:
            dec.extend(decompogen(args[i], symbol))
            args[i] = symbol
        result += [f.func(*args)] + list(uniq(dec))
```

Reply:
You'll have to import `uniq`; maybe an appropriate test would involve `u = 2*x + 3; decompogen(Max(u, u**2, u**3), x)`

Reply:
I have added a test `raises(TypeError, lambda: decompogen(Max(2*x + 3, (2*x + 3)**2, (2*x + 3)**3), x))` , Should a test be included for `decompogen(Min(5, x), x)` ?